sound/soc/sof/ipc4-topology.c

```diff
@@ -943,6 +944,21 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 
 	sof_ipc4_widget_update_kcontrol_module_id(swidget);
 
+	pipe_widget = swidget->spipe->pipe_widget;
+
+	/*
+	 * A Data Processing (DP) domain widget can be scheduled to run on a secondary core while
+	 * it's pipeline runs the primary core. In this case, the secondary core could get powered
+	 * off before the pipeline is deleted resulting in a crash when the firmware tries to free
+	 * the DP module during pipeline deletion. Prevent this by setting the pipe_widget's core
+	 * ID to match that of the widget so that the core would be kept powered on until the
```

Comment:
the pipe_widget->core is a concept that should not exist. The pipeline does not "run", only module instances are scheduled, and the scheduling does not happen at the pipeline level.

Reply:
Pipeline core ID was added to make FW management of pipeline easier; it is just an info for FW on which core it should run pipeline management code.

Reply:
> EDIT: and all LLs of this pipeline should run on the same core as pipeline itself (at least for now)

We already know that changes are coming for LL on multiple cores so we don't want to have to revisit the core management multiple times.... Let's do the things right and future-proof, shall we?